# HG changeset patch
# User mullan
# Date 1382735486 -3600
#      Fri Oct 25 22:11:26 2013 +0100
# Node ID 5b485ecccea9064e5036e1bb040fc2af1f773c20
# Parent  011d86bdcbf8d4fe0016351b775ff1f48f3620a1
7146431: java.security files out-of-sync
Reviewed-by: vinnie, xuelei, wetmore

diff -r 011d86bdcbf8 -r 5b485ecccea9 src/share/lib/security/java.security-solaris
--- openjdk/jdk/src/share/lib/security/java.security-solaris	Mon Jul 15 03:40:16 2013 -0700
+++ openjdk/jdk/src/share/lib/security/java.security-solaris	Fri Oct 25 22:11:26 2013 +0100
@@ -141,6 +141,7 @@
                com.sun.org.apache.xalan.internal.lib.,\
                com.sun.org.apache.xalan.internal.res.,\
                com.sun.org.apache.xalan.internal.templates.,\
+	       com.sun.org.apache.xalan.internal.utils.\
                com.sun.org.apache.xalan.internal.xslt.,\
                com.sun.org.apache.xalan.internal.xsltc.cmdline.,\
                com.sun.org.apache.xalan.internal.xsltc.compiler.,\
diff -r 011d86bdcbf8 -r 5b485ecccea9 src/share/lib/security/java.security-windows
--- openjdk/jdk/src/share/lib/security/java.security-windows	Mon Jul 15 03:40:16 2013 -0700
+++ openjdk/jdk/src/share/lib/security/java.security-windows	Fri Oct 25 22:11:26 2013 +0100
@@ -141,6 +141,7 @@
                com.sun.org.apache.xalan.internal.lib.,\
                com.sun.org.apache.xalan.internal.res.,\
                com.sun.org.apache.xalan.internal.templates.,\
+	       com.sun.org.apache.xalan.internal.utils.\
                com.sun.org.apache.xalan.internal.xslt.,\
                com.sun.org.apache.xalan.internal.xsltc.cmdline.,\
                com.sun.org.apache.xalan.internal.xsltc.compiler.,\
diff -r 011d86bdcbf8 -r 5b485ecccea9 test/java/lang/SecurityManager/CheckPackageAccess.java
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ openjdk/jdk/test/java/lang/SecurityManager/CheckPackageAccess.java	Fri Oct 25 22:11:26 2013 +0100
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2012, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 7146431
+ * @summary Test that internal JAXP packages cannot be accessed
+ */
+
+public class CheckPackageAccess {
+
+    public static void main(String[] args) throws Exception {
+
+        String[] pkgs = new String[] {
+            "com.sun.org.apache.xerces.internal.utils.",
+            "com.sun.org.apache.xalan.internal.utils." };
+        SecurityManager sm = new SecurityManager();
+        System.setSecurityManager(sm);
+        for (String pkg : pkgs) {
+            System.out.println("Checking package access for " + pkg);
+            try {
+                sm.checkPackageAccess(pkg);
+                throw new Exception("Expected SecurityException not thrown");
+            } catch (SecurityException se) { }
+        }
+    }
+}
